pallets/automation-time/src/tests.rs

```diff
@@ -2494,192 +2606,403 @@ fn trigger_tasks_completes_some_xcmp_tasks() {
+		let delegator = AccountId32::new(DELEGATOR_ACCOUNT);
+		get_funds(delegator.clone());
+		let before_balance = Balances::free_balance(delegator.clone());
+		// Minimum balance is half of the user's wallet balance
 		let account_minimum = before_balance / 2;
 		let frequency = 3_600;
 
 		let task_id = add_recurring_task_to_task_queue(
-			ALICE,
+			DELEGATOR_ACCOUNT,
 			vec![1],
 			SCHEDULED_TIME,
 			frequency,
 			Action::AutoCompoundDelegatedStake {
-				delegator: AccountId32::new(ALICE),
-				collator: AccountId32::new(BOB),
+				delegator: delegator.clone(),
+				collator: AccountId32::new(COLLATOR_ACCOUNT),
 				account_minimum,
 			},
+			vec![],
 		);
 
 		System::reset_events();
 
 		AutomationTime::trigger_tasks(Weight::from_ref_time(120_000));
 
-		events()
+		let emitted_events = events();
+
+		// Expected result:
+		// 1. Current execution will run
+		emitted_events
+			.clone()
 			.into_iter()
-			.find(|e| match e {
-				RuntimeEvent::AutomationTime(crate::Event::TaskScheduled { .. }) => true,
-				_ => false,
+			.find(|e| matches!(e, RuntimeEvent::AutomationTime(crate::Event::TaskExecuted { .. })))
+			.expect("TaskExecuted event should have been emitted");
+
+		// 2. Next execution will be scheduled
+		emitted_events
+			.clone()
+			.into_iter()
+			.find(|e| {
+				matches!(e, RuntimeEvent::AutomationTime(crate::Event::TaskRescheduled { .. }))
 			})
-			.expect("TaskScheduled event should have been emitted");
+			.expect("TaskRescheduled event should have been emitted");
+
 		let next_scheduled_time = SCHEDULED_TIME + frequency;
 		AutomationTime::get_scheduled_tasks(next_scheduled_time)
 			.expect("Task should have been rescheduled")
 			.tasks
 			.into_iter()
-			.find(|t| *t == (AccountId32::new(ALICE), task_id.clone()))
+			.find(|t| *t == (delegator.clone(), task_id.clone()))
 			.expect("Task should have been rescheduled");
-		let task = AutomationTime::get_account_task(AccountId32::new(ALICE), task_id.clone())
+		let task = AutomationTime::get_account_task(delegator.clone(), task_id)
 			.expect("Task should not have been removed from task map");
 		assert_eq!(task.schedule.known_executions_left(), 1);
 		assert_eq!(task.execution_times(), vec![next_scheduled_time]);
 
+		// 3. The task will re-run in the next execution time
 		Timestamp::set_timestamp(next_scheduled_time * 1_000);
-		get_funds(AccountId32::new(ALICE));
+		get_funds(delegator.clone());
 		System::reset_events();
 		AutomationTime::trigger_tasks(Weight::from_ref_time(100_000_000_000));
 
 		events()
 			.into_iter()
-			.find(|e| match e {
-				RuntimeEvent::AutomationTime(
-					crate::Event::SuccesfullyAutoCompoundedDelegatorStake { .. },
-				) => true,
-				_ => false,
-			})
-			.expect("AutoCompound success event should have been emitted");
+			.find(|e| matches!(e, RuntimeEvent::AutomationTime(crate::Event::TaskTriggered { .. })))
+			.expect("AutoCompoundDelegatorStakeSucceeded event should have been emitted again!");
```

Comment:
Please make sure all the expect messages updated. Here it shouldn’t be `AutoCompoundDelegatorStakeSucceeded` any more.